modules/Bio/EnsEMBL/IO/Translator/StructuralVariationFeature.pm

```diff
@@ -0,0 +1,339 @@
+              );                            
+#my %info_methods = ('AA' => 'ancestral_allele',
+#                    #'AC'  => 'allele_count',
+#                    #'AF'  => 'allele_frequency',
+#                    #'AN'  => 'total_allele_number',
+#                    'MA'  => 'minor_allele',
+#                    'MAF' => 'minor_allele_frequency',
+#                    'MAC' => 'minor_allele_count',
+#                    'NS'  => 'samples_with_data'
+#                   );
+
+=head2 new
+
+    Returntype   : Bio::EnsEMBL::IO::Translator::SampleGenotypeFeature
+
+=cut
+
+sub new {
+    my ($class, $info_list, $formats_list, $samples_list) = @_;
+  
+    my $self = $class->SUPER::new();
+
+    $samples_list = [] if (!$samples_list);
+    $self->{'samples_list'} = $samples_list;
+
+    if (scalar(@$samples_list) > 0) {
+      $field_callbacks{'format'} = 'format';
+    }
+    else {
+      warn "There is no individuals/samples list provided!\n";
+    }
+    
+    # Once we have the instance, add our customized callbacks
+    # to the translator
+    $self->add_callbacks(\%field_callbacks);
+    
+    $self->{'info_list'}    = $info_list;
+    $self->{'formats_list'} = $formats_list;
+
+    return $self;
+}
+
+
+sub slice_adaptor {
+  my $self = shift;
+  my $object = shift;
+
+  if (!$self->{'slice_adaptor'}) {
+    $self->{'slice_adaptor'} = $object->slice->adaptor
+  }
+  return $self->{'slice_adaptor'};
+}
+
+sub start {
+  my $self = shift;
+  my $object = shift;
+
+  return ( $object->seq_region_start() > $object->seq_region_end() ) ? $object->seq_region_end() : $object->seq_region_start();
+}
+
+sub end {
+  my $self = shift;
+  my $object = shift;
+  
+  return ( $object->seq_region_start() > $object->seq_region_end() ) ? $object->seq_region_start() : $object->seq_region_end();
+}
+
+
+#### Fields methods ####
+
+sub chrom {
+  my $self = shift;
+  my $object = shift;
+  
+  return $object->seq_region_name;
+}
+
+sub pos {
+  my $self = shift;
+  my $object = shift;
+
+  my $pos = $self->start($object);
+
+  # Get the alleles to check if we need to change the position
+  # For indel/insertion/deletion/unbalanced variants, we need to
+  # Get the ref base (and position) before the structural variant start
+  $pos--;
+
+  return $pos;
+}
+
+
+sub id {
+  my $self = shift;
+  my $object = shift;
+  
+  return $object->variation_name();
+}
+
+sub ref {
+  my $self = shift;
+  my $object = shift;
+  
+  my $ref = 'N';
+  
+  my $chrom = $self->chrom($object);
+  my $pos = $self->pos($object);
+
+  my $slice_adaptor = $self->slice_adaptor($object);
+  if ($slice_adaptor) {
+    my $slice = $slice_adaptor->fetch_by_toplevel_location("$chrom:$pos-$pos");
+    $ref = $slice->seq if defined($slice);
+  }
+  return $ref;
+}
+
+sub alt {
+  my $self = shift;
+  my $object = shift;
+
+  my $type = $object->class_SO_term;
+
+  return ($svtypes{$type}) ? '<'.$svtypes{$type}.'>' : '<SA>';
+}
+
+sub qual {
+  my $self = shift;
```

Comment:
At some point we should carve through these on an optimisation run to remove the useless lines of code.